### PR TITLE
closes cart page shouldn't show 'admin & handling' if it is zero #424

### PR DESCRIPTION
**Pull request by jtruong2**
#### What? Why?

The cart page should hide the table row 'admin & handling' if adjustments equal '$0.00'

#### What should we test?
Test should include 'shows admin & handling row if adjustments exist' and 'hides admin & handling if adjustments do not exist'

#### Release notes

Admin & Handling fee is only displayed in the cart if it is non-zero

### CODE REVIEW

File: app/views/spree/orders/_form.html.haml [hunk 1]
```diff
@@ -35,13 +35,13 @@
...
             %span.order-total.item-total= display_checkout_subtotal(@order)
           %td
-
-        %tr
-          %td.text-right{colspan:"3"}
-            = t :orders_form_admin
-          %td.text-right
-            %span.order-total.distribution-total= display_checkout_admin_and_handling_adjustments_total_for(@order)
-          %td
+        -if display_checkout_admin_and_handling_adjustments_total_for(@order)
```
Comment (sauloperez) [inline app/views/spree/orders/_form.html.haml]: Did you consider moving the conditional in `app/helpers/checkout_helper.rb:29` to here as:

```ruby
if display_checkout_admin_and_handling_adjustments_total_for(@order) == Spree::Money.new(0 , currency: @order.currency)
```

This way `display_checkout_admin_and_handling_adjustments_total_for` will keep returning just a data type, which keeps thing simple and reusable.

Comment (jtruong2) [reply app/views/spree/orders/_form.html.haml]: Okay thanks for pointing that out. I will make that change.

**Comment by sauloperez** [comment]
Rubocop is complaining. You can make him fix it for you with the `autocorrect` option.

```shell
$ bundle exec rubocop -a app/helpers/checkout_helper.rb
```

:ok_hand:

**Comment by sauloperez** [comment]
Watch out you have a spec failing :warning:

File: spec/helpers/checkout_helper_spec.rb [hunk 2]
```diff
@@ -30,4 +30,5 @@
     order.distributor.allow_guest_orders = false
     expect(helper.guest_checkout_allowed?).to be false
   end
+
```
Comment (oeoeaio) [inline spec/helpers/checkout_helper_spec.rb]: Remove this space please 🙏

File: app/helpers/checkout_helper.rb [hunk 3]
```diff
@@ -112,4 +113,5 @@ def payment_method_price(method, order)
       "{{ #{price} | localizeCurrency }}"
     end
   end
+
```
Comment (oeoeaio) [inline app/helpers/checkout_helper.rb]: These two spaces not required

**Comment by jtruong2** [comment]
@sauloperez @oeoeaio Sorry for the delayed response, I've been sick and away from the computer. I apologize but I won't be able to work on this issue in my current state. I'd like to free it up for others to fix and so I won't delay it for you all any longer. Thanks!

**Comment by oeoeaio** [comment]
No problems @jtruong2, thanks for letting us know. 🙏

**Comment by daniellemoorhead** [comment]
Closing this as no longer active and is attached to issue so can be found and reopened if work will start again.

**Pull request closed without merging**